packages/snaps-jest/src/types.ts

```diff
@@ -216,10 +216,54 @@ export type SnapPromptInterface = {
   cancel(): Promise<void>;
 };
 
+export type SnapDefaultInterfaceWithFooter = {
+  /**
+   * The content to show in the interface.
+   */
+  content: JSXElement;
+};
+
+export type SnapDefaultInterfaceWithPartialFooter = {
+  /**
+   * The content to show in the interface.
+   */
+  content: JSXElement;
+
+  /**
+   * Cancel the prompt.
+   */
+  cancel(): Promise<void>;
+};
+
+export type SnapDefaultInterfaceWithoutFooter = {
+  /**
+   * The content to show in the interface.
+   */
+  content: JSXElement;
+
+  /**
+   * Close the prompt.
+   *
+   * @param value - The value to close the prompt with.
+   */
+  ok(): Promise<void>;
+
+  /**
+   * Cancel the prompt.
+   */
+  cancel(): Promise<void>;
+};
```

Comment:
You could reduce the duplication here a little bit by reusing types.

```ts
type DefaultSnapInterfaceWithFooter = {
  content: JSXElement;
};

type DefaultSnapInterfaceWithPartialFooter = DefaultSnapInterfaceWithFooter & {
  cancel(): Promise<void>;
};

type DefaultSnapInterfaceWithoutFooter = DefaultSnapInterfaceWithPartialFooter & {
  ok(): Promise<void>;
};
```

Reply:
Might also be good to add JSDoc comments to each type to describe what the difference is.